lib/permanent_records/active_record_5_2.rb

```diff
@@ -0,0 +1,45 @@
+module HandlePermanentRecordsDestroyedInBelongsToAssociation
+  def handle_dependency
+    return unless load_target
+
+    case options[:dependent]
+    when :destroy
+      target.destroy
+      if target.respond_to?(:soft_destroyed?)
+        raise ActiveRecord::Rollback unless target.soft_destroyed?
+      elsif target.respond_to?(:destroyed?)
```

Comment:
also, seems like we only need to target `deleted?` since `permanent_records` handle the deleted as
```ruby
def deleted?
  if is_permanent?
    !!deleted_at # rubocop:disable Style/DoubleNegation
  else
    destroyed?
  end
end
```

Reply:
the `deleted?` method gets added to all AR models right?

Reply:
I assume so